src/mg_decoration.cpp

```diff
@@ -347,6 +347,16 @@ size_t DecoSchematic::generate(MMVManip *vm, PcgRandom *pr, v3s16 p)
 	if (!CONTAINS(c_place_on, c))
 		return 0;
 
+	// Check for a non-walkable node above, to be consistent with the
+	// behaviour of findGroundLevel and findLiquidSurface, which are
+	// used if the heightmap is not present. This avoids, for example,
+	// multiple tree trunks placed at the same position.
+	v3s16 em = vm->m_area.getExtent();
+	vm->m_area.add_y(em, vi, 1);
+	MapNode &n = vm->m_data[vi];
+	if (mg->ndef->get(n).walkable)
```

Comment:
This kind of hard coded logic leaves me scratching my head.  I'm not sure if checking the node above for walkable is really the right thing to do to begin with.